### PR TITLE
#338 Generate ci script

### LINKED ISSUE #338: Generate script for running PK in CI
Maven projects use PK via the included plugin. In order to run PK for other build systems like Go we need a different solution.

Proposal:
* PK generates a shell script that
  * Downloads PK from GitHub
  * Verifies the checksum
  * Runs it with `fix` or `verify`
* PK generates a GH workflow that runs this shell script
* PK updates the script for the latest version when run with "fix" -- similar to self update #140

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #338

### CODE REVIEW

File: project-keeper/src/main/java/com/exasol/projectkeeper/ProjectKeeper.java [hunk 1]
```diff
@@ -55,8 +55,15 @@ private ProjectKeeper(final Logger logger, final Path projectDir, final Path mvn
...
-        return new ProjectKeeper(logger, projectDir, mvnRepo, readConfig(projectDir), ownVersion);
+        return new ProjectKeeper(logger, projectDir, mvnRepo, readConfig(projectDir), getOwnVersion());
+    }
+
+    private static String getOwnVersion() {
+        final String packageVersion = ProjectKeeper.class.getPackage().getImplementationVersion();
+        if (packageVersion != null) {
+            return packageVersion;
+        }
+        return "(unknownVersion)";
```
Comment (kaklakariada) [inline project-keeper/src/main/java/com/exasol/projectkeeper/ProjectKeeper.java:66]: This avoids a Nullpointer Exception when running the tests from Eclipse

File: project-keeper/src/main/java/com/exasol/projectkeeper/validators/files/FileTemplateFromResource.java [hunk 2]
```diff
@@ -48,6 +48,6 @@ public Path getPathInProject() {
 
     @Override
     public String getContent() {
-        return new ResourceReader().readFromResource("templates/" + this.pathInProject);
+        return new ResourceReader().readFromResource(this.pathToTemplate);
```
Comment (kaklakariada) [inline project-keeper/src/main/java/com/exasol/projectkeeper/validators/files/FileTemplateFromResource.java:51]: This was a bug

File: project-keeper/src/main/java/com/exasol/projectkeeper/validators/files/FileTemplatesFactory.java [hunk 3]
```diff
@@ -32,6 +34,8 @@ List<FileTemplate> getGlobalTemplates(final List<AnalyzedSource> sources) {
                 templates.add(new FileTemplateFromResource(
                         ".github/workflows/release_droid_release_on_maven_central.yml", REQUIRE_EXACT));
             }
+        } else if (onlyGolangProjects(sources)) {
+            templates.addAll(getGolangTemplates());
```
[the remark this answers is not in the document]
Comment (ckunki) [reply project-keeper/src/main/java/com/exasol/projectkeeper/validators/files/FileTemplatesFactory.java:38]: I can create a branch / PR if you agree.

File: project-keeper/src/main/resources/non_maven_templates/.github/workflows/project-keeper.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+set -o pipefail
+
+readonly pk_mode="${1-verify}";
+readonly version="##VERSION##"
+
+readonly pk_jar="$HOME/.m2/repository/com/exasol/project-keeper-cli/$version/project-keeper-cli-$version.jar"
+
+if [ ! -f "$pk_jar" ]; then
+    echo "Downloading Using Project Keeper $version"
+    mvn --batch-mode org.apache.maven.plugins:maven-dependency-plugin:3.3.0:get -Dartifact=com.exasol:project-keeper-cli:$version
```
Comment (ckunki) [inline project-keeper/src/main/resources/non_maven_templates/.github/workflows/project-keeper.sh:14]: maybe on interactive use you want to run mvn without `--batch-mode`?
should we move cmd line arg `--batch-mode` to the yml workflow file?

**Review by ckunki — CHANGES_REQUESTED** [review]
Only some minor proposal / questions regarding wording and logic.
the "templates/" prefix is still unclear to me. 
Happy to talk about it 😄

**Review by ckunki — APPROVED** [review]

**Pull request merged**